svc-aggregation/system/aggregate.go

```diff
@@ -93,37 +96,154 @@ func (e *ExternalInterface) CreateAggregate(req *aggregatorproto.AggregatorReque
-	return response.RPC{
-		StatusCode: http.StatusNotImplemented,
+	aggregate, err := agmodel.GetAggregate(req.URL)
+	if err != nil {
+		log.Printf("error getting  Aggregate : %v", err)
+		errorMessage := err.Error()
+		if errors.DBKeyNotFound == err.ErrNo() {
+			return common.GeneralError(http.StatusNotFound, response.ResourceNotFound, err.Error(), []interface{}{"Aggregate", req.URL}, nil)
+		}
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
+	}
+	var data = strings.Split(req.URL, "/redfish/v1/AggregationService/Aggregates/")
+	commonResponse := response.Response{
+		OdataType:    "#Aggregate.v1_0_0.Aggregate",
+		OdataID:      req.URL,
+		OdataContext: "/redfish/v1/$metadata#Aggregate.Aggregate",
+		ID:           data[1],
+		Name:         "Aggregate",
+	}
+	var resp = response.RPC{
+		StatusCode:    http.StatusOK,
+		StatusMessage: response.Success,
+	}
+	resp.Header = map[string]string{
+		"Cache-Control":     "no-cache",
+		"Connection":        "keep-alive",
+		"Content-type":      "application/json; charset=utf-8",
+		"Transfer-Encoding": "chunked",
+		"OData-Version":     "4.0",
 	}
+	commonResponse.CreateGenericResponse(response.Success)
+	resp.Body = agresponse.AggregateResponse{
+		Response: commonResponse,
+		Elements: aggregate.Elements,
+	}
+	return resp
 }
 
 // DeleteAggregate is the handler for deleting an aggregate
 // if the aggregate id is present then delete from the db else return an error.
 func (e *ExternalInterface) DeleteAggregate(req *aggregatorproto.AggregatorRequest) response.RPC {
-	// TODO add functionality to create an aggregate
-	return response.RPC{
-		StatusCode: http.StatusNotImplemented,
+	var resp response.RPC
+	_, err := agmodel.GetAggregate(req.URL)
+	if err != nil {
+		log.Printf("error getting  Aggregate : %v", err)
+		errorMessage := err.Error()
+		if errors.DBKeyNotFound == err.ErrNo() {
+			return common.GeneralError(http.StatusNotFound, response.ResourceNotFound, err.Error(), []interface{}{"Aggregate", req.URL}, nil)
+		}
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
 	}
+	err = agmodel.DeleteAggregate(req.URL)
+	if err != nil {
+		log.Printf("error while deleting an aggregate : %v", err)
+		errorMessage := err.Error()
+		if errors.DBKeyNotFound == err.ErrNo() {
+			return common.GeneralError(http.StatusNotFound, response.ResourceNotFound, err.Error(), []interface{}{"Aggregate", req.URL}, nil)
+		}
+		return common.GeneralError(http.StatusInternalServerError, response.InternalError, errorMessage, nil, nil)
+	}
+	var data = strings.Split(req.URL, "/redfish/v1/AggregationService/Aggregates/")
+	commonResponse := response.Response{
+		OdataType:    "#Aggregate.v1_0_0.Aggregate",
+		OdataID:      req.URL,
+		OdataContext: "/redfish/v1/$metadata#Aggregate.Aggregate",
+		ID:           data[1],
+		Name:         "Aggregate",
+	}
+
+	resp.StatusCode = http.StatusOK
```

Comment:
change status code to http.StatusNoContent

Reply:
updated